assets/js/script.js

```diff
@@ -0,0 +1,84 @@
+// Assignment Code
+var generateBtn = document.querySelector("#generate");
+
+// Write password to the #password input
+function writePassword() {
+  // new secure password
+  const password = generatePassword();
+  const passwordText = document.querySelector("#password");
+
+  passwordText.value = password;
+}
+
+function generatePassword() {
+  // prompt for the length of the password
+  const passLength = prompt(
+    "Please enter how many characters you would like your password to be. \nMust be between 8 - 128 characters."
+  );
+
+  const lengthConfirm = parseInt(passLength);
+  console.log(lengthConfirm);
+
+  // choose a length of at least 8 characters and no more than 128 characters
+  if (passLength < 8 || passLength > 128 || isNaN(passLength)) {
+    alert("Error. Choice is not valid. Please try again.");
+    return;
+  }
+
+  const collectionOfValids = [];
+  const upperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("");
+  const lowerCase = "abcdefghiklmnopqrsttuvwxtz".split("");
+  const number = "0123456789".split("");
+  const special = "*&^%$#@!?><{}".split("");
+
```

Comment:
As mentioned in the feedback, I think it would of been more beneficial to have these as arrays from the start and then just pushing them into an empty array when they are called. This does seem to work though, just be careful with `for` loops.

Reply:
I have taken note and had 1;1 with tutor.